### PR TITLE
Set segment only when cookie diverges

### PR DESCRIPTION
**Pull request by mcandeia**

### CODE REVIEW

File: vtex/middleware.ts [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
+/**
+ * by default segment starts with null values
+ */
+const DEFAULT_SEGMENT: Partial<Segment> = {
+  utmi_campaign: null,
+  utm_campaign: null,
+  utm_source: null,
+};
+
+const SEGMENT_ONCE_KEY = Symbol("Set segment on context");
```
Comment (lucis) [inline vtex/middleware.ts:20]: Why do we need `Symbol` here? I remember briefly

Either way, WDYT about adding a comment about it?

**Review by lucis — APPROVED** [review]
Do you need help testing it? I think Dudu from ZeeDog can help you getting a valid `utm_segment`

File: vtex/middleware.ts (hunk 1 above)
Comment (mcandeia) [reply vtex/middleware.ts:20]: We are using WeakMap for bag @lucis Symbols are the only possible way for indexing WeakMaps. 

https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/WeakMap

I can certainly add a comment about it linking to the `WeakMap` docs, how about that?

Anyways I think this should not be necessary as the type of bag is weakmaps which should be addressed with documentation

Comment (mcandeia) [reply vtex/middleware.ts:20]: About testing: I think i did not change the behavior of segment itself, just not setting it twice when it's already set

**Pull request merged**